api/src/main/resources/Java/libraries/okhttp-gson/api.mustache

```diff
@@ -0,0 +1,277 @@
+            Set<ConstraintViolation<{{classname}}>> violations = executableValidator.validateParameters(this, method,
+                    parameterValues);
+
+            if (violations.size() == 0) {
+                com.squareup.okhttp.Call {{localVariablePrefix}}call = {{operationId}}Call({{#allParams}}{{paramName}}, {{/allParams}}progressListener, progressRequestListener);
+                return {{localVariablePrefix}}call;
+
+            } else {
+                throw new BeanValidationException((Set) violations);
+            }
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
+            throw new ApiException(e.getMessage());
+        } catch (SecurityException e) {
+            e.printStackTrace();
+            throw new ApiException(e.getMessage());
+        }
+
+        {{/performBeanValidation}}
+
+
+
+
+    }
+
+    /**
+     * {{summary}}
+     * {{notes}}{{#allParams}}
+     * @param {{paramName}} {{description}}{{#required}} (required){{/required}}{{^required}} (optional{{#defaultValue}}, default to {{{.}}}{{/defaultValue}}){{/required}}{{/allParams}}{{#returnType}}
+     * @return {{returnType}}{{/returnType}}
+     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
+     */
+    public {{#returnType}}{{{returnType}}} {{/returnType}}{{^returnType}}void {{/returnType}}{{operationId}}({{#allParams}}{{{dataType}}} {{paramName}}{{#hasMore}}, {{/hasMore}}{{/allParams}}) throws ApiException {
+        {{! Begin AoU-specific additions }}
+        // Generate an OpenCensus tracing span for this API method call.
+        Scope scope = tracer.spanBuilderWithExplicitParent("{{classname}}.{{operationId}}", tracer.getCurrentSpan())
+            .setSpanKind(Span.Kind.CLIENT).startScopedSpan();
+        Span span = tracer.getCurrentSpan();
+        // Some of these attribute keys use naming from the OpenCensus standard. See
+        // https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md
+        span.putAttribute("package", AttributeValue.stringAttributeValue("{{package}}"));
+        span.putAttribute("http.host", AttributeValue.stringAttributeValue(apiClient.getBasePath()));
+        span.putAttribute("http.path", AttributeValue.stringAttributeValue("{{path}}"));
+        span.putAttribute("http.method", AttributeValue.stringAttributeValue("{{httpMethod}}"));
+        span.putAttribute("description", AttributeValue.stringAttributeValue("{{summary}}"));
+        {{#returnType}}span.putAttribute("responseType", AttributeValue.stringAttributeValue("{{{returnType}}}"));{{/returnType}}
+        span.putAttribute("connectionTimeout", AttributeValue.longAttributeValue(apiClient.getConnectTimeout()));
+
+        try {
```

Comment:
Can you do a try-with-resources block here on `scope`?

Reply:
It's tempting, but I don't think it would significantly add to the clarity here. There were also some pretty scary warnings about try-with-resources on these scoped spans, which may affect the error-handling case: https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/trace/SpanBuilder.java#L236